### PR TITLE
User posts query

### LINKED ISSUE #104: Create model function to get the posts of a user

### PR DESCRIPTION
**Pull request by gruvw**
A new query functionality on the model aspect that allows any viewmodel to retrieve the lists of posts of a given user.

This functionality will prove itself particularly useful for the viewmodel of the profile page, where we want to display all the posts of a given user.

Fixes #104

### CODE REVIEW

**Comment by gruvw** [comment]
BTW, I also think that there is a problem here as the tests are passing locally but not on the CI... idk why yet, I will investigate this issue.
Meanwhile, can somebody confirm that all the tests on this branch pass on their machine ? ty

**Comment by yoannLafore** [comment]
On my machine, the tests run correctly with `flutter test`.
However, I get the same error as the CI when I run the workflow with `act`.

File: test/models/database/post/mock_post_data.dart [hunk 1]
```diff
@@ -25,6 +27,30 @@ class MockPostFirestore {
...
+      ),
+      location: PostLocationFirestore(
+        geoPoint: location,
+        geohash: point.hash,
+      ),
+      data: PostData(
+        ownerId: userId,
+        title: "title",
+        description: "desciption",
+        publicationTime: Timestamp.now(),
```
Comment (yoannLafore) [inline test/models/database/post/mock_post_data.dart:48]: I just think that we may want to create the timestamp from a predefined value.
Like that its value doesn't change between runs and the tests operate on the same data.
```suggestion
        publicationTime: Timestamp.fromMicrosecondsSinceEpoch(1000000),
```

File: test/services/database/post_repository_test.dart [hunk 2]
```diff
@@ -213,5 +214,36 @@ void main() {
...
+
+      expect(actualPosts1.length, 2);
+      expect(actualPosts1.length, actualPosts2.length);
+
+      for (final post in actualPosts1) {
+        expect(post.data.ownerId.value, userId1.value);
+      }
+      for (final post in actualPosts2) {
+        expect(post.data.ownerId.value, userId2.value);
+      }
```
Comment (yoannLafore) [inline test/services/database/post_repository_test.dart:246]: Nice test :+1: 
I believe we can make it a little more concise like that:

```suggestion
      expect(actualPosts1, unorderedEquals([postsData1, postsData2]));
      expect(actualPosts2, unorderedEquals([postsData3, postsData4]));
```

**Review by yoannLafore — APPROVED** [review]
The PR looks good to me :+1: 
I have just made two little suggestions below.

Overall, great job :tada: 
Thanks for your good work !

File: test/models/database/post/mock_post_data.dart (hunk 1 above)
Comment (gruvw) [reply test/models/database/post/mock_post_data.dart]: Applied in f6c7890

File: test/services/database/post_repository_test.dart (hunk 2 above)
Comment (gruvw) [reply test/services/database/post_repository_test.dart]: Applied in b06a70f

**Review by CHOOSEIT — APPROVED** [review]
Clean, great job !

LGTM !

**Pull request merged**